src/transformers/models/patchtst/modeling_patchtst.py

```diff
@@ -0,0 +1,1958 @@
+                )  # Add: residual connection with residual dropout
+            else:
+                ## Multi-Head attention and Add residual connection and Norm
+                src = self.norm_sublayer2(
+                    src + self.dropout_path2(self.self_attn(src)[0])
+                )  # src: [(bs*sequence_length) x nvars x d_model]
+            src = (
+                src.reshape(bs, sequence_length, num_input_channels, d_model).transpose(1, 2).contiguous()
+            )  # src: [bs x nvars x sequence_length x d_model]
+
+        # Third sublayer: mixing across hidden
+        src = src.view(
+            bs * num_input_channels, sequence_length, d_model
+        )  # src: [(bs*nvars) x sequence_length x d_model]
+        if self.pre_norm:
+            ## Norm and Position-wise Feed-Forward and Add residual connection
+            src = src + self.dropout_path3(
+                self.ff(self.norm_sublayer3(src))
+            )  # Add: residual connection with residual dropout
+        else:
+            ## Position-wise Feed-Forward and Add residual connection and Norm
+            src = self.norm_sublayer3(
+                src + self.dropout_path3(self.ff(src))
+            )  # Add: residual connection with residual dropout
+        src = src.reshape(bs, num_input_channels, sequence_length, d_model)  # [bs x nvars x sequence_length x d_model]
+
+        return src
+
+
+class PatchTSTPreTrainedModel(PreTrainedModel):
+    config_class = PatchTSTConfig
+    base_model_prefix = "model"
+    main_input_name = "past_values"
+    supports_gradient_checkpointing = False
+
+    def _init_weights(self, module):
+        """Initialize weights"""
+        if self.config.use_cls_token:
+            torch.nn.init.normal_(self.config.cls_token, std=0.02)
+        elif isinstance(module, nn.LayerNorm):
+            module.bias.data.zero_()
+            module.weight.data.fill_(1.0)
+        elif isinstance(module, (nn.Linear, nn.Conv1d)):
+            module.weight.data.normal_(mean=0.0, std=self.config.init_std)
+            if module.bias is not None:
+                module.bias.data.zero_()
+
+    def _set_gradient_checkpointing(self, module, value=False):
+        if isinstance(module, (ChannelAttentionPatchTSTEncoder)):
+            module.gradient_checkpointing = value
+
+
+class ChannelAttentionPatchTSTEncoder(PatchTSTPreTrainedModel):
+    def __init__(self, config: PatchTSTConfig):
+        super().__init__(config)
+        self.num_input_channels = config.num_input_channels
+        self.num_patches = config.num_patches
+        self.patch_length = config.patch_length
+        self.d_model = config.d_model
+        self.shared_embedding = config.shared_embedding
+        self.use_cls_token = config.use_cls_token
+        self.gradient_checkpointing = False
+
+        # Input encoding: projection of feature vectors onto a d-dim vector space
+        if not config.shared_embedding:
+            self.w_p = nn.ModuleList()
```

Comment:
w_p is not the best variable name

Reply:
Thank you for the suggestion. It is changed to input_embedding.